### PR TITLE
feat: Introduce metrics.URL parameter

### PR DESCRIPTION
**Pull request by bradbeam**
This introduces the metrics.URL parameter to allow a more explicit
targeting of a remote data source.

Decided to implement a local http test server for fetching the prometheus metric so we can remove the external dependency.

Signed-off-by: Brad Beam <brad.beam@carbonrelay.com>

### CODE REVIEW

**Review by andrewhare — APPROVED** [review]

File: api/v1beta1/experiment_types.go [hunk 1]
```diff
@@ -117,6 +117,10 @@ type Metric struct {
 	Port intstr.IntOrString `json:"port,omitempty"`
 	// URL path component used to collect the metric value from an endpoint (used as a prefix for the Prometheus API)
 	Path string `json:"path,omitempty"`
+	// URL to query for fetching metrics.
```
Comment (jgustie) [inline api/v1beta1/experiment_types.go:120]: Do we want to just remove these from `v1beta1`?

Comment (jgustie) [reply api/v1beta1/experiment_types.go:120]: (that didn't highlight properly, by "these" I meant all the other URL like fields)

Comment (bradbeam) [reply api/v1beta1/experiment_types.go:120]: Yeah we can, wanted to try to keep the scope of this minimal to make it easier to review :)

**Review by jgustie — APPROVED** [review]

**Comment by jgustie** [comment]
I'm waiting to get client registration deployed, at which time 1.6.1 will get released, can we hold off on merging until then? If it's urgent I can release the patch version off a branch.

**Pull request merged**